The job **`mingw-check-2`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/15658595391/job/44113010720) [(plain)](https://github.com/rust-lang/rust/commit/940b7d262ad769cd2f9fe2a8465c373697b563d3/checks/44113010720/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

error[E0658]: use of unstable library feature `file_lock`
   --> src/tools/miri/src/shims/unix/fs.rs:107:17
    |
107 |             Err(TryLockError::Error(err)) => interp_ok(Err(err)),
    |                 ^^^^^^^^^^^^^^^^^^^
    |
    = note: see issue #130994 <https://github.com/rust-lang/rust/issues/130994> for more information
    = help: add `#![feature(file_lock)]` to the crate attributes to enable
    = note: this compiler was built on 2025-05-25; consider upgrading it if it is out of date

error[E0658]: use of unstable library feature `file_lock`
   --> src/tools/miri/src/shims/unix/fs.rs:108:17
    |
108 |             Err(TryLockError::WouldBlock) =>
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: see issue #130994 <https://github.com/rust-lang/rust/issues/130994> for more information
    = help: add `#![feature(file_lock)]` to the crate attributes to enable
    = note: this compiler was built on 2025-05-25; consider upgrading it if it is out of date

error[E0658]: use of unstable library feature `file_lock`
  --> src/tools/miri/src/shims/unix/fs.rs:98:54
   |
98 |             SharedLock { nonblocking } => (self.file.try_lock_shared(), nonblocking),
   |                                                      ^^^^^^^^^^^^^^^
   |
   = note: see issue #130994 <https://github.com/rust-lang/rust/issues/130994> for more information
   = help: add `#![feature(file_lock)]` to the crate attributes to enable
   = note: this compiler was built on 2025-05-25; consider upgrading it if it is out of date

error[E0658]: use of unstable library feature `file_lock`
  --> src/tools/miri/src/shims/unix/fs.rs:99:57
   |
99 |             ExclusiveLock { nonblocking } => (self.file.try_lock(), nonblocking),
   |                                                         ^^^^^^^^
   |
   = note: see issue #130994 <https://github.com/rust-lang/rust/issues/130994> for more information
   = help: add `#![feature(file_lock)]` to the crate attributes to enable
   = note: this compiler was built on 2025-05-25; consider upgrading it if it is out of date

error[E0658]: use of unstable library feature `file_lock`
   --> src/tools/miri/src/shims/unix/fs.rs:101:44
    |
101 |                 return interp_ok(self.file.unlock());
    |                                            ^^^^^^
    |
    = note: see issue #130994 <https://github.com/rust-lang/rust/issues/130994> for more information
    = help: add `#![feature(file_lock)]` to the crate attributes to enable
    = note: this compiler was built on 2025-05-25; consider upgrading it if it is out of date

error[E0658]: use of unstable library feature `file_lock`
   --> src/tools/miri/src/shims/unix/fs.rs:107:37
    |
107 |             Err(TryLockError::Error(err)) => interp_ok(Err(err)),
    |                                     ^^^
    |
    = note: see issue #130994 <https://github.com/rust-lang/rust/issues/130994> for more information
    = help: add `#![feature(file_lock)]` to the crate attributes to enable
    = note: this compiler was built on 2025-05-25; consider upgrading it if it is out of date
```

</details>

--- a/library/std/src/fs.rs
+++ b/library/std/src/fs.rs
@@ -121,7 +121,7 @@ pub struct File {
 ///
 /// [`try_lock`]: File::try_lock
 /// [`try_lock_shared`]: File::try_lock_shared
-#[unstable(feature = "file_lock", issue = "130994")]
+#[stable(feature = "file_lock", since = "CURRENT_RUSTC_VERSION")]
 pub enum TryLockError {
     /// The lock could not be acquired due to an I/O error on the file. The standard library will
     /// not return an [`ErrorKind::WouldBlock`] error inside [`TryLockError::Error`]
@@ -366,10 +366,10 @@ pub fn write<P: AsRef<Path>, C: AsRef<[u8]>>(path: P, contents: C) -> io::Result
     inner(path.as_ref(), contents.as_ref())
 }
 
-#[unstable(feature = "file_lock", issue = "130994")]
+#[stable(feature = "file_lock", since = "CURRENT_RUSTC_VERSION")]
 impl error::Error for TryLockError {}
 
-#[unstable(feature = "file_lock", issue = "130994")]
+#[stable(feature = "file_lock", since = "CURRENT_RUSTC_VERSION")]
 impl fmt::Debug for TryLockError {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         match self {
@@ -379,7 +379,7 @@ impl fmt::Debug for TryLockError {
     }
 }
 
-#[unstable(feature = "file_lock", issue = "130994")]
+#[stable(feature = "file_lock", since = "CURRENT_RUSTC_VERSION")]
 impl fmt::Display for TryLockError {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         match self {
@@ -390,7 +390,7 @@ impl fmt::Display for TryLockError {
     }
 }
 
-#[unstable(feature = "file_lock", issue = "130994")]
+#[stable(feature = "file_lock", since = "CURRENT_RUSTC_VERSION")]
 impl From<TryLockError> for io::Error {
     fn from(err: TryLockError) -> io::Error {
         match err {
@@ -713,7 +713,6 @@ impl File {
     /// # Examples
     ///
     /// ```no_run
-    /// #![feature(file_lock)]
     /// use std::fs::File;
     ///
     /// fn main() -> std::io::Result<()> {
@@ -722,7 +721,7 @@ impl File {
     ///     Ok(())
     /// }
     /// ```
-    #[unstable(feature = "file_lock", issue = "130994")]
+    #[stable(feature = "file_lock", since = "CURRENT_RUSTC_VERSION")]
     pub fn lock(&self) -> io::Result<()> {
         self.inner.lock()
     }
@@ -766,7 +765,6 @@ impl File {
     /// # Examples
     ///
     /// ```no_run
-    /// #![feature(file_lock)]
     /// use std::fs::File;
     ///
     /// fn main() -> std::io::Result<()> {
@@ -775,7 +773,7 @@ impl File {
     ///     Ok(())
     /// }
     /// ```
-    #[unstable(feature = "file_lock", issue = "130994")]
+    #[stable(feature = "file_lock", since = "CURRENT_RUSTC_VERSION")]
     pub fn lock_shared(&self) -> io::Result<()> {
         self.inner.lock_shared()
     }
@@ -824,7 +822,6 @@ impl File {
     /// # Examples
     ///
     /// ```no_run
-    /// #![feature(file_lock)]
     /// use std::fs::{File, TryLockError};
     ///
     /// fn main() -> std::io::Result<()> {
@@ -840,7 +837,7 @@ impl File {
     ///     Ok(())
     /// }
     /// ```
-    #[unstable(feature = "file_lock", issue = "130994")]
+    #[stable(feature = "file_lock", since = "CURRENT_RUSTC_VERSION")]
     pub fn try_lock(&self) -> Result<(), TryLockError> {
         self.inner.try_lock()
     }
@@ -888,7 +885,6 @@ impl File {
     /// # Examples
     ///
     /// ```no_run
-    /// #![feature(file_lock)]
     /// use std::fs::{File, TryLockError};
     ///
     /// fn main() -> std::io::Result<()> {
@@ -905,7 +901,7 @@ impl File {
     ///     Ok(())
     /// }
     /// ```
-    #[unstable(feature = "file_lock", issue = "130994")]
+    #[stable(feature = "file_lock", since = "CURRENT_RUSTC_VERSION")]
     pub fn try_lock_shared(&self) -> Result<(), TryLockError> {
         self.inner.try_lock_shared()
     }
@@ -933,7 +929,6 @@ impl File {
     /// # Examples
     ///
     /// ```no_run
-    /// #![feature(file_lock)]
     /// use std::fs::File;
     ///
     /// fn main() -> std::io::Result<()> {
@@ -943,7 +938,7 @@ impl File {
     ///     Ok(())
     /// }
     /// ```
-    #[unstable(feature = "file_lock", issue = "130994")]
+    #[stable(feature = "file_lock", since = "CURRENT_RUSTC_VERSION")]
     pub fn unlock(&self) -> io::Result<()> {
         self.inner.unlock()
     }

--- a/src/tools/miri/src/lib.rs
+++ b/src/tools/miri/src/lib.rs
@@ -16,7 +16,7 @@
 #![feature(unqualified_local_imports)]
 #![feature(derive_coerce_pointee)]
 #![feature(arbitrary_self_types)]
-#![feature(file_lock)]
+#![cfg_attr(bootstrap, feature(file_lock))]
 // Configure clippy and other lints
 #![allow(
     clippy::collapsible_else_if,

--- a/src/tools/miri/tests/pass/shims/fs.rs
+++ b/src/tools/miri/tests/pass/shims/fs.rs
@@ -2,7 +2,6 @@
 
 #![feature(io_error_more)]
 #![feature(io_error_uncategorized)]
-#![feature(file_lock)]
 
 use std::collections::BTreeMap;
 use std::ffi::OsString;

